app/controllers/case_court_reports_controller.rb

```diff
@@ -14,15 +14,16 @@ def index
   # GET /case_court_reports/:id
   def show
     authorize CaseCourtReport
-    unless @casa_case
+    if !@casa_case || !@casa_case.court_report.attached?
       flash[:alert] = "Report #{params[:id]} is not found."
       redirect_to(case_court_reports_path) and return # rubocop:disable Style/AndOr
     end
 
     respond_to do |format|
       format.docx do
-        report_data = generate_report_to_string(@casa_case)
-        send_report(report_data)
+        @casa_case.court_report.open do |file|
```

Comment:
`@casa_case.court_report.open` will yield a `Tempfile` from the active storage blob. We can use this `Tempfile` path in `send_data`

https://apidock.com/rails/ActiveStorage/Blob/open